src/Integration/ProfileConflicts/ConflictsManager.cs

```diff
@@ -0,0 +1,187 @@
+using System.Linq;
+
+namespace SonarLint.VisualStudio.Integration.ProfileConflicts
+{
+    /// <summary>
+    /// Conflicts manager for a SonarQube bound solution
+    /// </summary>
+    internal class ConflictsManager : IConflictsManager
+    {
+        private readonly IServiceProvider serviceProvider;
+
+        public ConflictsManager(IServiceProvider serviceProvider)
+        {
+            if (serviceProvider == null)
+            {
+                throw new ArgumentNullException(nameof(serviceProvider));
+            }
+
+            this.serviceProvider = serviceProvider;
+        }
+
+        public IReadOnlyList<ProjectRuleSetConflict> GetCurrentConflicts()
+        {
+            // Note: some of the assumptions (see asserts below) are because have just bounded the solution (as documented on the interface),
+            // in other cases assuming that the rule set are indeed on disk is not possible, and in fact resyncing 
+            // would be required when we have missing rulesets, otherwise finding conflicts will not be possible.
+
+            RuleSetAggregate[] aggregatedRuleSets = GetAggregatedSolutionRuleSets();
+
+            if (aggregatedRuleSets.Length > 0)
+            {
+                return FindConflicts(aggregatedRuleSets);
+            }
+
+            return new ProjectRuleSetConflict[0];
+        }
+
+        private IReadOnlyList<ProjectRuleSetConflict> FindConflicts(RuleSetAggregate[] aggregatedRuleSet)
+        {
+            List<ProjectRuleSetConflict> conflicts = new List<ProjectRuleSetConflict>();
+
+            IRuleSetInspector inspector = this.serviceProvider.GetService<IRuleSetInspector>();
+            inspector.AssertLocalServiceIsNotNull();
+
+            // At the moment single threaded, if needed this could be done in parallel
+            foreach (RuleSetAggregate aggregate in aggregatedRuleSet)
+            {
+                try
+                {
+                    RuleConflictInfo conflict = inspector.FindConflictingRules(aggregate.BaselineFilePath, aggregate.RuleSetFilePath, aggregate.RuleSetDirectories);
+                    Debug.Assert(conflict != null);
+
+                    if (conflict?.HasConflicts ?? false)
+                    {
+                        conflicts.Add(new ProjectRuleSetConflict(conflict, aggregate));
+                    }
+                }
+                catch (Exception ex)
+                {
+                    if (ErrorHandler.IsCriticalException(ex))
+                    {
+                        throw;
+                    }
+
+                    this.WriteWarning(Strings.ConflictsManagerFailedInFindConflicts, aggregate.RuleSetFilePath, aggregate.BaselineFilePath, ex.Message);
+                    Debug.Fail("Failed to resolve conflict for " + aggregate.RuleSetFilePath, ex.ToString());
+                }
+            }
+
+            return conflicts;
+        }
+
+        private RuleSetAggregate[] GetAggregatedSolutionRuleSets()
```

Comment:
We don't want to analyze and fix the same rule more than once and we don't want to show duplicate information in the summary (will be confusing).